api/turing/validation/validator.go

```diff
@@ -129,6 +129,43 @@ func checkTrafficRuleName(sl validator.StructLevel, fieldName string, value stri
 	}
 }
 
+func checkTrafficRuleCount(sl validator.StructLevel, fieldName string, rules models.TrafficRules) {
+	ruleLength := len(rules)
+	ruleCountDescription := "There should be either 0 or more than 1 traffic rules specified."
+	if ruleLength == 1 {
+		sl.ReportError(rules, fieldName, "trafficRule", ruleCountDescription, "")
+	}
+	// Check for presence of Default Traffic Rule
+	hasDefaultRule := false
+	noDefaultRuleDescription := "Default rule should be provided."
+	if ruleLength > 1 {
+		for _, rule := range rules {
+			if rule.Name == "default" {
+				hasDefaultRule = true
+			}
+		}
+		// Default rule should be present if any custom rules exist
+		if !hasDefaultRule {
+			sl.ReportError(rules, fieldName, "trafficRule", noDefaultRuleDescription, "")
+		}
+	}
+}
+
+func checkTrafficRuleConditions(sl validator.StructLevel, fieldName string, rule *models.TrafficRule) {
+	noConditionsDefaultRuleDescription := "Default rule should not have any conditions specified."
+	noConditionsCustomRuleDescription := "Custom rule must have conditions specified."
+
+	if rule.Name == "default" {
+		if len(rule.Conditions) != 0 {
+			sl.ReportError(rule, fieldName, "trafficRule", noConditionsDefaultRuleDescription, "")
+		}
+	} else {
+		if len(rule.Conditions) == 0 {
+			sl.ReportError(rule, fieldName, "trafficRule", noConditionsCustomRuleDescription, "")
+		}
+	}
+}
+
```

Comment:
In general, all of the special handling around the default rule is making me think we may be better off splitting it into a new field or redoing the `traffic_rules` column to explicitly state in the contract. Do you have any ideas around this?

Reply:
Also, do we have an end-user migration strategy for this?

Reply:
This PR is still pending 2 items as listed in the description's `TO-DO` section. I'm thinking to wrap this PR up with just the DB migration script, and leaving the default traffic rule information UI for the next PR since we're revamping the Router Details view and doing this now might be a waste of effort. wdyt?

Reply:
I think the data migration is best coupled with the change that introduces the schema changes. If not, when other commits need to be deployed internally and this change is included, some workflows may be broken.

But I understand this PR has already gotten pretty big, so I agree with separating the migration script into another PR, if it will take some time to sort out. Something to keep in mind for the future.

Less concerned about the second bullet on the UI changes. On that note, we also need to update the user docs but that could go with the new UI card.

Reply:
👍🏻 Aligned with just including DB migration script for this PR and leaving UI changes for the next.